### PR TITLE
Restores the stopped threads after a restoreState request

### LINKED ISSUE #205: Restore the full state
We should now be able to restore the full state of the current execution using the `stoppedThreads` field from the `debugInfo` reply:

https://github.com/jupyterlab/debugger/blob/b1b41742d13aab5c6243d7fca0c66f91ac288ba9/src/tokens.ts#L231

The debugger will then need to request the state of variables for each of the threads.

### PR DESCRIPTION
**Pull request by JohanMabille**
Fixes #205

### CODE REVIEW

File: src/service.ts [hunk 1]
```diff
@@ -165,10 +165,10 @@ export class DebugService implements IDebugger {
   }
 
   /**
-   * Whether the current thread is stopped.
+   * Whether there exist a thread in stopped stated.
```
Comment (jtpio) [inline src/service.ts]: Typo?

```suggestion
   * Whether there exist a thread in a stopped state.
```

File: src/service.ts [hunk 2]
```diff
@@ -165,10 +165,10 @@ export class DebugService implements IDebugger {
...
    */
-  isThreadStopped(): boolean {
-    return this._stoppedThreads.has(this.currentThread());
+  hasThreadStopped(): boolean {
```
Comment (jtpio) [inline src/service.ts]: Or maybe `hasStoppedThread()`?

File: src/debugger.ts [hunk 3]
```diff
@@ -156,6 +156,14 @@ export namespace Debugger {
       this._modeChanged.emit(mode);
     }
 
+    get stoppedThreads(): Set<number> {
```
Comment (jtpio) [inline src/debugger.ts]: Let's also add a docstring?

File: src/service.ts (hunk 2 above)
Comment (JohanMabille) [reply src/service.ts]: yep that would be consistent with the name of the variable

**Comment by jtpio** [comment]
This is what it looks like:

restore-stopped-threads

**Comment by jtpio** [comment]
But reloading the page:

restore-stopped-threads-reload

**Comment by JohanMabille** [comment]
The issue relative to reloading the page is deeper and requires that we take some decision regarding the different options we have to handle it.
I will open an issue to give more detail about that.

File: src/service.ts.orig [hunk 4]
```diff
@@ -0,0 +1,543 @@
+// Copyright (c) Jupyter Development Team.
```
Comment (jtpio) [inline src/service.ts.orig]: Should we add these `.orig` to the `.gitignore`?

Comment (JohanMabille) [reply src/service.ts.orig]: Yes we can do that. I'm going to remove the .orig added by mistake, I'll add the gitignore change in a dedicated commit.

**Pull request merged**